dspace-api/src/main/java/org/dspace/handle/HandlePlugin.java

```diff
@@ -254,6 +254,54 @@ public void scanNAs(ScanCallback callback) throws HandleException {
     // Resolving methods
     ////////////////////////////////////////
 
+    /**
+     * Resolve the given handle to DSpace object.
+     *
+     * @param context the context
+     * @param handle the handle to resolve
+     * @return the resolved DSpaceObject
+     * @throws HandleException if an error occurs during resolution
+     */
+    private static DSpaceObject resolveHandleToObject(Context context, String handle) throws HandleException {
+        try {
+            return handleClarinService.resolveToObject(context, handle);
+        } catch (Exception e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception in resolveHandleToObject", e);
+            }
+            throw new HandleException(HandleException.INTERNAL_ERROR);
+        }
+    }
+
+    /**
+     * Retrieves handle values as a map.
+     *
+     * @param handle the handle to resolve
+     * @return a map containing the handle values
+     * @throws HandleException if an error occurs during handle resolution
+     */
+    public static Map<String, String> getMapHandleValues(String handle) throws HandleException {
+        if (log.isInfoEnabled()) {
+            log.info("Called getMapHandleValues");
+        }
+        loadServices();
+        Context context = new Context();
+        try {
+            DSpaceObject dso = null;
+            boolean resolveMetadata = configurationService.getBooleanProperty("lr.pid.resolvemetadata", true);
```

Comment:
?

Reply:
But the resul of extraMetadata form null is empty map  not null https://github.com/dataquest-dev/DSpace/pull/761/files#diff-89c959e468673dbb5fad73c3dadd9ec90a76cf12f16066a989450e4d5ed85c76L572

Reply:
but on line 290 you init it with null

Reply:
when dso is null, the return value is a `new LinkedHashMap<>()`, not `null`. But we will get this information in `extractMetadata`. Based on that, I believe it’s better to leave it as it is now.